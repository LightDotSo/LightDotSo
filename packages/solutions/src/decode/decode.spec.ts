// Copyright (C) 2023 Light, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import { expect, test } from "vitest";
import { decodeSignature } from ".";
import { toBytes } from "viem";

test("decode", () => {
  const sampleSignature1 =
    "0x0001636911b800019fa7b7e8ed25088c413074818ac10ab3bbcddb120bbec85083f3ba254e5547d953fe615a6474fd365326244dedd7afa3911ad39c956ca096d721064d6b29055d1b02";

  const topology = decodeSignature(toBytes(sampleSignature1));

  expect(topology).toMatchSnapshot();
});
