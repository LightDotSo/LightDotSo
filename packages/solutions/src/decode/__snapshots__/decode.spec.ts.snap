// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`decode 1`] = `
{
  "body": {
    "checkpoint": 1667830200n,
    "threshold": 1n,
    "tree": {
      "signature": "0x9fa7b7e8ed25088c413074818ac10ab3bbcddb120bbec85083f3ba254e5547d953fe615a6474fd365326244dedd7afa3911ad39c956ca096d721064d6b29055d1b02",
      "weight": 1n,
    },
  },
  "type": 0,
}
`;
